src/nbl/asset/utils/ISPIRVOptimizer.cpp

```diff
@@ -76,8 +112,15 @@ nbl::core::smart_refctd_ptr<ICPUBuffer> ISPIRVOptimizer::optimize(const uint32_t
 
     spvtools::Optimizer opt(SPIRV_VERSION);
 
-    for (E_OPTIMIZER_PASS pass : m_passes)
-        opt.RegisterPass(create_pass_f[pass]());
+    for (E_OPTIMIZER_PASS pass : m_passes) {
+        if (getSpirvOptimizerPass(pass) != nullptr)
+        {
+            opt.RegisterPass(create_pass_f[pass]());
+        } else
+        {
+            logger.log("Optimizer pass is unknown or not supported!", system::ILogger::ELL_WARNING);
+        }
```

Comment:
whats going on here, why not use `getSpirvOptoimizerPass(pass)` directly?

Reply:
so that we can get rid of `create_pass_f` ? https://github.com/Devsh-Graphics-Programming/Nabla/pull/839#discussion_r2035368381